In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')
import warnings; warnings.simplefilter('ignore')

df=pd.read_csv("c19new.csv")

df.head()
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
df[['Cured', 'Confirmed Cases']] =scaler.fit_transform(df[['Cured', 'Confirmed Cases']])  
df.tail()

Date     Cured  Deaths  Confirmed Cases
267  23-10-2020  2.679081  117306         2.475919
268  24-10-2020  2.711539  117956         2.497821
269  25-10-2020  2.741368  118534         2.518393
270  26-10-2020  2.769769  119014         2.536921
271  27-10-2020  2.800446  119502         2.551888

In [3]:
X = df[['Confirmed Cases','Cured']]

In [4]:
y = df['Deaths']

In [5]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)

In [6]:
from sklearn.svm import SVR

In [7]:
svr = SVR().fit(X_train, y_train)

In [8]:
from sklearn import metrics

In [9]:
y_pred = svr.predict(X_test)
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test, y_pred))

Mean Absolute Error: 28117.979812848993


In [10]:
print('Mean Squared Error:', metrics.mean_squared_error(y_test, y_pred))

Mean Squared Error: 1760145046.7256756


In [11]:
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))

Root Mean Squared Error: 41954.08259902337


In [12]:
reg=svr

print('Support Vector Regression: R^2 score on test set', reg.score(X_test, y_test)*100)

Support Vector Regression: R^2 score on test set -27.947561160468016


<h3>Evaluation Metrics using Support Vector Regression</h3><br> 
<p>Mean Absolute Error: 128117.979812848993 <br> 
Mean Squared Error: 1760145046.7256756 <br>
Root Mean Squared Error: 41954.08259902337 <br>
R^2 score on test set: -27.947561160468016</p>